featuretools_tsfresh_primitives/utils.py

```diff
@@ -1,45 +1,75 @@
-import numpy as np
-import pandas as pd
+from featuretools.primitives import AggregationPrimitive, TransformPrimitive
+from tsfresh.feature_extraction.settings import ComprehensiveFCParameters
 
 import featuretools_tsfresh_primitives
 
 
-def _pascal_case(snake_str):
-    words = snake_str.split('_')
-    titled_words = map(str.title, words)
-    return ''.join(titled_words)
+def comprehensive_fc_parameters():
+    """It is basically a dictionary, which is a mapping from string to a dictionary list of parameters.
 
+    Returns:
+        parameters (dict) : a dictionary list of parameters
 
-def _tsfresh_to_featuretools(fc_name):
-    return getattr(featuretools_tsfresh_primitives, _pascal_case(fc_name))
+    Docstring source:
+    https://tsfresh.readthedocs.io/en/latest/api/tsfresh.feature_extraction.html#tsfresh.feature_extraction.settings.ComprehensiveFCParameters
+    """
+    parameters = ComprehensiveFCParameters()
+
+    # when a partial autocorrelation has a lag of zero,
+    # an error is raised from `tsfresh.feature_extraction.extract_features`
+    partial_autocorrelation = parameters['partial_autocorrelation']
+    for index, values in enumerate(partial_autocorrelation):
+        if values['lag'] == 0: del partial_autocorrelation[index]
+
+    return parameters
 
 
-def primitives_from_fc_settings(fc_settings):
+def supported_primitives():
+    """Generates the currently supported primitives.
+
+        Returns:
+            generator: primitive classes
     """
-    Return a list of :class:AggregationPrimitive from tsfresh settings.
-    The format is the same as the argument `default_fc_parameters` of
-    :func:`tsfresh.feature_extraction.extract_features`
+    types = AggregationPrimitive, TransformPrimitive
+    for key in dir(featuretools_tsfresh_primitives):
+        value = getattr(featuretools_tsfresh_primitives, key)
+        is_primitive = isinstance(value, type) and issubclass(value, types)
+        if is_primitive: yield value
+
+
+def primitives_from_fc_settings(fc_settings=None):
+    """Return a list of :class:AggregationPrimitive from tsfresh settings.
+    The format is the same as the argument `default_fc_parameters` of :func:`tsfresh.feature_extraction.extract_features`
```

Comment:
We should add a line mentioning that if no settings are passed, the default tsfresh settings are used